modules/core/src/main/scala/codec/TemporalCodecs.scala

```diff
@@ -13,83 +13,139 @@ import java.time.LocalDateTime
+    temporal(localDateTimeFormatter(6), LocalDateTime.parse, Type.timestamp)
 
   def timestamp(precision: Int): Codec[LocalDateTime] =
-    precision match {
-      case 0 => temporal("yyyy-MM-dd HH:mm:ss",        LocalDateTime.parse, Type.timestamp(0))
-      case 1 => temporal("yyyy-MM-dd HH:mm:ss.S",      LocalDateTime.parse, Type.timestamp(1))
-      case 2 => temporal("yyyy-MM-dd HH:mm:ss.SS",     LocalDateTime.parse, Type.timestamp(2))
-      case 3 => temporal("yyyy-MM-dd HH:mm:ss.SSS",    LocalDateTime.parse, Type.timestamp(3))
-      case 4 => temporal("yyyy-MM-dd HH:mm:ss.SSSS",   LocalDateTime.parse, Type.timestamp(4))
-      case 5 => temporal("yyyy-MM-dd HH:mm:ss.SSSSS",  LocalDateTime.parse, Type.timestamp(5))
-      case 6 => temporal("yyyy-MM-dd HH:mm:ss.SSSSSS", LocalDateTime.parse, Type.timestamp(6))
-      case _ => throw new IllegalArgumentException(s"timestamp($precision): invalid precision, expected 0-6")
+    if(precision >= 0 && precision <= 6) {
+      temporal(localDateTimeFormatter(precision), LocalDateTime.parse, Type.timestamp(precision))
+    } else {
+      throw new IllegalArgumentException(s"timestamp($precision): invalid precision, expected 0-6")
     }
-
+    
   val timestamptz: Codec[OffsetDateTime] =
-    temporal("yyyy-MM-dd HH:mm:ss.SSSSSSx", OffsetDateTime.parse, Type.timestamptz)
-
-  def timestamptz(precision: Int): Codec[LocalDateTime] =
-    precision match {
-      case 0 => temporal("yyyy-MM-dd HH:mm:ssx",        LocalDateTime.parse, Type.timestamptz(0))
-      case 1 => temporal("yyyy-MM-dd HH:mm:ss.Sx",      LocalDateTime.parse, Type.timestamptz(1))
-      case 2 => temporal("yyyy-MM-dd HH:mm:ss.SSx",     LocalDateTime.parse, Type.timestamptz(2))
-      case 3 => temporal("yyyy-MM-dd HH:mm:ss.SSSx",    LocalDateTime.parse, Type.timestamptz(3))
-      case 4 => temporal("yyyy-MM-dd HH:mm:ss.SSSSx",   LocalDateTime.parse, Type.timestamptz(4))
-      case 5 => temporal("yyyy-MM-dd HH:mm:ss.SSSSSx",  LocalDateTime.parse, Type.timestamptz(5))
-      case 6 => temporal("yyyy-MM-dd HH:mm:ss.SSSSSSx", LocalDateTime.parse, Type.timestamptz(6))
-      case _ => throw new IllegalArgumentException(s"timestamptz($precision): invalid precision, expected 0-6")
+    temporal(offsetDateTimeFormatter(6), OffsetDateTime.parse, Type.timestamptz)
+
+  def timestamptz(precision: Int): Codec[OffsetDateTime] =
+    if(precision >= 0 && precision <= 6) {
+      temporal(offsetDateTimeFormatter(precision), OffsetDateTime.parse, Type.timestamptz(precision))
+    } else {
+      throw new IllegalArgumentException(s"timestampz($precision): invalid precision, expected 0-6")
     }
 
-  // todo: intervals, which will need a new data type
+  private def intervalCodec(tpe: Type): Codec[Duration] = Codec.simple(
+    duration => duration.toString,
+    str => Either.catchOnly[DateTimeParseException](Duration.parse(str)).leftMap(_.toString),
+    tpe
+  )
```

Comment:
Wait, what? Java's `Duration` toString/parse just happen to match Postgres input/output formats?

Reply:
Luckily, it does ;) Duration#toString uses ISO-8601 format, so we have postgres interval input as ISO-8601. We also configure postgres to use ISO-8601 interval output here: https://github.com/tpolecat/skunk/blob/724fb5570e73091406704e6c24a0af4aa395750d/modules/core/src/main/scala/net/message/StartupMessage.scala#L37

Reply:
Although I don't take the precision into consideration, so somebody writing `sql"'PT5.6666'::interval(6)".query(duration(2))` might expect to get back `PT5.66` but will get `PT5.6666` instead. I will figure out if there was any reason I left it this way and fix if not.

Reply:
Oh, the above query is actually not possible to execute, because skunk will check that `interval(6)` and `interval(2)` are not the same type and throw an error. So we do not need to trim parts of seconds to right precision when reading from postgres - the precision here will always be right. 

What would still make sense is trimming the strings to the right precision when **writing** to postgres, i.e. when writing with codec `interval(2)` we would trim `PT5.6666` to `PT5.66`. This doesn't change any semantics, just would save a few bytes. Do you think it's worth doing?